### PR TITLE
[CPP] Separate common utils out from llm_chat.cc

### PR DESCRIPTION
**Pull request by MasterJH5574**
This PR separates out the tokenizer creation function, the random number generator out from `llm_chat.cc` as a preparation step for batching inference support, since these functions/modules are also used in the same way in batching inference.

### CODE REVIEW

File: cpp/rng.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+#ifndef MLC_LLM_RNG_H_
+#define MLC_LLM_RNG_H_
+
+#include <random>
+
+namespace mlc {
+namespace llm {
+
+// Random number generator
+class RandomGenerator {
```
Comment (tqchen) [inline cpp/rng.h:16]: random.h

Comment (MasterJH5574) [reply cpp/rng.h]: Updated

**Comment by MasterJH5574** [comment]
Ah looks like I messed something up during rebase. Will resolve it later

**Comment by MasterJH5574** [comment]
Okay it's back to life now.

**Review by tqchen — APPROVED** [review]

File: cpp/tokenizer.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ *  Copyright (c) 2023 by Contributors
+ * \file tokenizer.h
+ * \brief Header of tokenizer related functions.
+ */
+
+#ifndef MLC_LLM_DLL
+#ifdef _WIN32
+#ifdef MLC_LLM_EXPORTS
+#define MLC_LLM_DLL __declspec(dllexport)
+#else
```
Comment (tqchen) [inline cpp/tokenizer.h:11]: This should go to base.h

File: cpp/tokenizer.h [hunk 3]
```diff
@@ -0,0 +1,34 @@
+/*!
+ *  Copyright (c) 2023 by Contributors
+ * \file tokenizer.h
+ * \brief Header of tokenizer related functions.
```
Comment (tqchen) [inline cpp/tokenizer.h:4]: tokenizers.h

File: cpp/tokenizer.cc [hunk 4]
```diff
@@ -0,0 +1,61 @@
+/*!
+ *  Copyright (c) 2023 by Contributors
+ * \file tokenizer.cc
+ */
+
```
Comment (tqchen) [inline cpp/tokenizer.cc:5]: tokenizers.cc

File: cpp/common.h [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+#ifndef MLC_LLM_COMMON_H_
+#define MLC_LLM_COMMON_H_
+
+#include <fstream>
+#include <string>
+
+namespace mlc {
+namespace llm {
+
+inline std::string LoadBytesFromFile(const std::string& path) {
```
Comment (tqchen) [inline cpp/common.h:16]: support.h

**Pull request merged**